modules/core/src/components/declarative-ui/xviz-metric.js

```diff
@@ -125,32 +127,42 @@ class XVIZMetricComponent extends PureComponent {
       formatValue,
       horizontalGridLines,
       verticalGridLines,
-      getColor
+      getColor,
+      streams
     } = this.props;
     const isLoading = currentTime == null; /* eslint-disable-line no-eq-null, eqeqeq */
     const timeDomain = Number.isFinite(startTime) ? [startTime, endTime] : null;
+    const {timeSeries} = this.state;
+    const missingStreams = streams.filter(streamToDisplay => !timeSeries.data[streamToDisplay]);
```

Comment:
I think this should actually be in the state, set in componentWillReceiveProps. Otherwise you have to recalculate this value every re-render.

Reply:
Moved.